File: lib/backend/models/auction_model.js
const mongoose = require('mongoose');

const auctionSchema = mongoose.Schema(
    {
        uid: {
            type: String,
            required: true,
        },
        aid: {
            type: String,
            required: true,
        },
        title: {
            type: String,
            required: true,
        },
        des: {
            type: String,
            required: true,
        },
        startingPrice: {
            type: Number,
            required: true,
        },
        numberOfUnits: {
            type: Number,
            required: true,
        },
        currentPrice: {
            type: Number,
            required: true,
        },
        urls: [
                {
                    type: String,
                    require: true,
                },
        ],
        bidList: [
            {
                name: {
                    type: String,
                require: true,
                },
                uid: {
                    type: String,
                require: true,
                },
                bidPrice: {
                    type: Number,
                require: true,
                },
            },
    ],
    }
);

const schema = mongoose.model('auctionSchema', auctionSchema);

module.exports = schema;

File: lib/models/auction_model.dart
import 'bid_model.dart';

class AuctionModel{
  String title, des, uid, aid;
  int numberOfUnits, startingPrice, currentPrice;
  List<dynamic> urls;
  List bidList;

  AuctionModel({
    required this.title,
    required this.uid,
    required this.aid,
    required this.startingPrice,
    required this.des,
    required this.numberOfUnits,
    required this.currentPrice,
    required this.urls,
    required this.bidList,
  });
}